src/execution_plan/ops/op_update.c

```diff
@@ -149,6 +150,26 @@ static void _CommitUpdates(OpUpdate *op) {
     if(op->result_set) op->result_set->stats.properties_set += op->pending_updates_count;
 }
 
+/* We only cache records if op_update is not the last
+ * operation within the execution-plan, which means
+ * others operations might inspect thoese records.
+ * Example: MATCH (n) SET n.v=n.v+1 RETURN n */
+static inline bool _ShouldCacheRecord(OpUpdate *op) {
+    return (op->op.parent != NULL);
+}
+
+static Record _handoff(OpUpdate* op) {
+    Record r = NULL;    
+    if(op->records) {
+        uint records_count = array_len(op->records);
+        /* TODO: poping a record out of op->records 
+         * will reverse the order in which records 
+         * are passed down the execution plan. */
```

Comment:
I can't think of a scenario in which this poses a problem - do you have any in mind?

Reply:
Maybe where an index scan is involved and an order by is specified